inference-engine/src/auto_plugin/auto_exec_network.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <map>
+#include <unordered_map>
+
+#include "ie_metric_helpers.hpp"
+#include "auto_exec_network.hpp"
+#include "auto_infer_request.hpp"
+
+namespace AutoPlugin {
+    using namespace InferenceEngine;
+
+AutoExecutableNetwork::AutoExecutableNetwork(const ExecutableNetwork& network,
+                                             const DeviceInformation& deviceInfo,
+                                             const bool               needPerfCounters) :
+    _deviceInfo(deviceInfo),
+    _network(network),
+    _config(deviceInfo.config.begin(), deviceInfo.config.end()),
+    _needPerfCounters(needPerfCounters) {
+}
+
+AutoExecutableNetwork::~AutoExecutableNetwork() = default;
+
+RemoteContext::Ptr AutoExecutableNetwork::GetContext() const {
+    try {
+        return _network.GetContext();
+    } catch (const NotImplemented&) {
+        IE_THROW(NotImplemented) << "None of the devices in the AUTO has an associated remote context.";
+    }
+}
+
+IInferRequestInternal::Ptr AutoExecutableNetwork::CreateInferRequestImpl(InputsDataMap networkInputs,
+                                                                         OutputsDataMap networkOutputs) {
+    auto inferRequest = _network.CreateInferRequest();
+    return std::make_shared<AutoInferRequest>(networkInputs, networkOutputs, inferRequest);
+}
+
+IInferRequestInternal::Ptr AutoExecutableNetwork::CreateInferRequest() {
```

Comment:
do we need to override this method? I suppose default implementation is the same

Reply:
yes, will delete this.